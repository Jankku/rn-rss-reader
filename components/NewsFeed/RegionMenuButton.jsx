import { Pressable } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

function RegionMenuButton({ onPress, color }) {
  return (
    <Pressable onPress={onPress} testID="RegionMenuButton">
      <Ionicons name={'funnel'} size={24} color={color} />
    </Pressable>
  );
}

export default RegionMenuButton;
