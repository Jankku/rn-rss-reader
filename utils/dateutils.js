import { DateTime } from 'luxon';

/**
 * For articles released today return time and date otherwise.
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date
 */
export function RFC2822ToTimeOrDate(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate =
    dateObject.toRelativeCalendar() === 'today'
      ? dateObject.toLocaleString(DateTime.TIME_24_SIMPLE)
      : dateObject.toLocaleString(DateTime.DATE_SHORT);
  return formattedDate;
}

/**
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date in DateTime format
 */
export function RFC2822ToDateTime(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate = dateObject.toLocaleString(DateTime.DATETIME_SHORT);
  return formattedDate;
}
