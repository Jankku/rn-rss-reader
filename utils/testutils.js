import { render } from '@testing-library/react-native';
import { ThemeContextProvider } from '../context/ThemeContext';
import { LocationContextProvider } from '../context/LocationContext';
import { RegionContextProvider } from '../context/RegionContext';

const AllProviders = ({ children }) => {
  return (
    <LocationContextProvider>
      <RegionContextProvider>
        <ThemeContextProvider>{children}</ThemeContextProvider>
      </RegionContextProvider>
    </LocationContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
