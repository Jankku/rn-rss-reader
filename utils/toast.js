import Toast from 'react-native-root-toast';

const showToast = (message) =>
  Toast.show(message, {
    position: 100,
    opacity: 0.9,
    shadow: false,
  });

export default showToast;
