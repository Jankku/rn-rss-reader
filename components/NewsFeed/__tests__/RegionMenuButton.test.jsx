import RegionMenuButton from '../RegionMenuButton';
import { render, fireEvent } from '../../../utils/testutils';

describe('Test RegionMenuButton', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(<RegionMenuButton onPress={() => {}} color={'#000'} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should call onPress', () => {
    const onPress = jest.fn();
    const { getByTestId } = render(<RegionMenuButton onPress={onPress} color={'#000'} />);
    const button = getByTestId('RegionMenuButton');

    fireEvent.press(button);
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
