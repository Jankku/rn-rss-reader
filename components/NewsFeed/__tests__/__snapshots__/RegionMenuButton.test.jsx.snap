// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test RegionMenuButton Should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="RegionMenuButton"
>
  <Text />
</View>
`;
